src/components/Timeline/src/StepIcon.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { StylesProvider } from '@material-ui/core';
+import SvgIcon, { SvgIconProps, CheckCircleIcon } from '@gemeente-denhaag/icons';
+import { stepIconClasses } from './bem-mapping';
+import clsx from 'clsx';
+import { TimelineComponent } from './types';
+
+export interface StepIconProps extends TimelineComponent {
+  /**
+   * The label displayed in the step icon.
+   */
+  icon?: number;
+}
+
+/**
+ * Displays an icon in a Step component within a Stepper component.
+ * @param props The properties of a StepIcon component.
+ * @constructor Constructs an instance of StepIcon.
+ */
+export const StepIcon: React.FC<StepIconProps> = (props: StepIconProps) => {
+  let color: SvgIconProps['color'] = 'primary';
+  if (props.disabled) {
+    color = 'disabled';
+  } else if (props.active) {
+    color = 'action';
+  } else if (props.completed) {
+    color = 'secondary';
+  }
+
+  const textClasses = clsx('denhaag-timeline__step-icon__text', {
+    'denhaag-timeline__step-icon__text--active': props.active,
+    'denhaag-timeline__step-icon__text--completed': props.completed,
+  });
```

Comment:
You can't chain together multiple "Elements" in BEM. It is always one block, followed by at most 1 element and 1 modifier. In this case this should be: `denhaag-timeline__step-icon-text`, `denhaag-timeline__step-icon-text--active`, etc. If you feel like this results in unreadable class names: it is allowed to use single underscores in the name of the element and modifier (NOT in the block name though!). So we could consider `denhaag-timeline__step-icon_text`.

Reply:
done

Reply:
You're saying done, and yet i'm still seeing two '__' in the classname 😋

Reply:
Sorry, I forgot to push my changes